packages/core/src/chains/configureChains.test.ts

```diff
@@ -0,0 +1,336 @@
+        ).toMatchInlineSnapshot('"wss://mainnet.infura.io/ws/v3/infuraId"')
+        expect(
+          webSocketProvider({ chainId: chain.polygon.id }).connection.url,
+        ).toMatchInlineSnapshot(
+          '"wss://polygon-mainnet.infura.io/ws/v3/infuraId"',
+        )
+      })
+
+      it('throws an error if Infura does not support a chain', () => {
+        expect(() =>
+          configureChains(
+            [
+              chain.mainnet,
+              chain.polygon,
+              chain.optimism,
+              chain.arbitrum,
+              chain.localhost,
+            ],
+
+            [infuraProvider(infuraId)],
+          ),
+        ).toThrowErrorMatchingInlineSnapshot(`
+          "Could not find valid API provider configuration for chain \\"Localhost\\".
+
+          You may need to add \`fallbackProvider\` to \`configureChains\` so that the chain can fall back to the public RPC URL."
+        `)
+      })
+    })
+
+    describe('fallback', () => {
+      it('populate with fallback configuration if all chains have a default RPC URL', () => {
+        const { chains, provider } = configureChains(
+          [chain.mainnet, chain.polygon, chain.optimism, chain.arbitrum],
+          [fallbackProvider()],
+        )
+
+        expect(chains.map((chain) => chain.rpcUrls.default))
+          .toMatchInlineSnapshot(`
+          [
+            "https://eth-mainnet.alchemyapi.io/v2/_gg7wSSi0KMBsdKnGVfHDueq6xMB9EkC",
+            "https://polygon-rpc.com",
+            "https://mainnet.optimism.io",
+            "https://arb1.arbitrum.io/rpc",
+          ]
+        `)
+
+        expect(
+          provider({ chainId: chain.mainnet.id }).connection.url,
+        ).toMatchInlineSnapshot(
+          '"https://eth-mainnet.alchemyapi.io/v2/_gg7wSSi0KMBsdKnGVfHDueq6xMB9EkC"',
+        )
+        expect(
+          provider({ chainId: chain.polygon.id }).connection.url,
+        ).toMatchInlineSnapshot('"https://polygon-rpc.com"')
+      })
+
+      it('throws an error if a chain does not have a default RPC URL', () => {
+        const polygon = { ...chain.polygon, rpcUrls: { default: '' } }
+
+        expect(() =>
+          configureChains(
+            [
+              chain.mainnet,
+              polygon,
+              chain.optimism,
+              chain.arbitrum,
+              chain.localhost,
+            ],
+
+            [fallbackProvider()],
+          ),
+        ).toThrowErrorMatchingInlineSnapshot(`
+          "Could not find valid API provider configuration for chain \\"Polygon\\".
+
+          You may need to add \`fallbackProvider\` to \`configureChains\` so that the chain can fall back to the public RPC URL."
+        `)
+      })
+    })
+
+    describe('jsonRpc', () => {
+      beforeAll(() => {
+        ;[chain.mainnet, chain.polygon, chain.optimism, chain.arbitrum].forEach(
+          (chain) => {
+            nock(`https://${chain.id}.example.com`).get('/').reply(200, {})
```

Comment:
can we either use `msw` here or convert `useEnsAvatar` and `fetchEnsAvatar` to `nock`?